lxc/storage.go

```diff
@@ -460,17 +460,42 @@ func (c *cmdStorageInfo) Run(cmd *cobra.Command, args []string) error {
 	poolusedby[usedbystring] = map[string][]string{}
 
 	/* Build up the usedby map
-	/1.0/{instances,images,profiles}/storagepoolname
+	/1.0/{instances,images,profiles}/storagepoolname and
+	/1.0/storage-pools/<poolname>/<type>/<volname>
 	remove the /1.0/ and build the map based on the resources name as key
 	and resources details as value */
 	for _, v := range pool.UsedBy {
-		bytype := string(strings.Split(v[5:], "/")[0])
-		bywhat := string(strings.Split(v[5:], "/")[1])
+		u, err := url.Parse(v)
+		if err != nil {
+			continue
+		}
+
+		fields := strings.Split(u.Path[5:], "/")
+		bytype := fields[0]
+		bywhat := fields[1]
+
+		if bytype == "storage-pools" {
+			bytype = "volumes"
+			bywhat = fields[4]
+		}
+
+		var info string
+
+		// Show info regarding the project and target if present.
+		for k, v := range u.Query() {
```

Comment:
Does this section need to check its dealing with either project or target query args?
Also do we need to consider if the query args are returned bin a different order, that would change the output of info right?

Reply:
> Does this section need to check its dealing with either project or target query args?

No, it just takes what it gets. And I believe that's either project or target.

> Also do we need to consider if the query args are returned bin a different order, that would change the output of info right?

A different order would change the output, but LXD always puts the target first.

Reply:
Thanks, thats clearer than before also as you've added labels to the info fields. :)